src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java

```diff
@@ -158,6 +162,37 @@ public void createTableWithOnlyMetadataColumn(final String tableName) {
     LOGGER.info("Created table {} with only RECORD_METADATA column", tableName);
   }
 
+  @Override
+  public void initializeMetadataColumnTypeForIceberg(String tableName) {
+    checkConnection();
+    InternalUtils.assertNotEmpty("tableName", tableName);
+    String query =
+        "ALTER ICEBERG TABLE identifier(?) ALTER COLUMN RECORD_METADATA SET DATA TYPE OBJECT("
+            + "offset INTEGER,"
```

Comment:
double-check if the list of properties is correct and the data types

Reply:
Looks good, but I wonder if we should add `LogAppendTime BIGINT` as well. I found the following code
```
// ignore if no timestamp
    if (record.timestampType() != TimestampType.NO_TIMESTAMP_TYPE
        && metadataConfig.createtimeFlag) {
      meta.put(record.timestampType().name, record.timestamp());
    }
```

and there are also tests like `public void testTimeStamp() throws IOException {`

Reply:
done